lib/SILOptimizer/Utils/CanonicalOSSALifetime.cpp

```diff
@@ -87,21 +91,30 @@ static void diagnose(ASTContext &Context, SourceLoc loc, Diag<T...> diag,
   Context.Diags.diagnose(loc, diag, std::forward<U>(args)...);
 }
 
-/// The lifetime extends beyond given consuming use. Copy the value.
-///
-/// This can set the operand value, but cannot invalidate the use iterator.
-void swift::copyLiveUse(Operand *use, InstModCallbacks &instModCallbacks) {
-  SILInstruction *user = use->getUser();
-  SILBuilderWithScope builder(user->getIterator());
+static DestroyValueInst *dynCastToDestroyOf(SILInstruction *instruction,
+                                            SILValue def) {
+  auto *destroy = dyn_cast<DestroyValueInst>(instruction);
+  if (!destroy)
+    return nullptr;
+  auto originalDestroyedDef = destroy->getOperand();
+  if (originalDestroyedDef == def)
+    return destroy;
+  auto underlyingDestroyedDef =
+      CanonicalizeOSSALifetime::getCanonicalCopiedDef(originalDestroyedDef);
+  if (underlyingDestroyedDef != def)
+    return nullptr;
+  return destroy;
+}
 
-  auto loc = RegularLocation::getAutoGeneratedLocation(user->getLoc());
-  auto *copy = builder.createCopyValue(loc, use->get());
-  instModCallbacks.createdNewInst(copy);
-  use->set(copy);
+namespace swift {
+struct CanonicalizeOSSALifetimeBoundary {
```

Comment:
Not a big deal, but does there need to be two types of lifetime boundaries? Aren't the conceptually and practically identical? You can just assert that there are no dead defs if those can't occur.

Reply:
Yeah, that's fair.  Switched over to PrunedLivenessBoundary.